From d2dc9481d2af25b035778c67eaf0bfd2d2c59dd8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20=C4=8Ciha=C5=99?= <michal@cihar.com>
Date: Fri, 29 Apr 2016 15:28:32 +0200
Subject: [PATCH] Fix possible path existence disclossure
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

On non released versions (where line counts are not precalculated) it
was possible to check for file existence due to limited checks for
supplied path.

Signed-off-by: Michal Čihař <michal@cihar.com>
---
 libraries/error_report.lib.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libraries/error_report.lib.php b/libraries/error_report.lib.php
index 1d7741e816c..856425606f9 100644
--- a/libraries/error_report.lib.php
+++ b/libraries/error_report.lib.php
@@ -238,7 +238,7 @@ function PMA_countLines($filename)
     foreach (explode('/', $filename) as $part) {
         if ($part == '..') {
             $depath--;
-        } elseif ($part != '.') {
+        } elseif ($part != '.' || $part === '') {
             $depath++;
         }
         if ($depath < 0) {
